### PR TITLE
fix: AI playing twice when AI marker is X

### PR DESCRIPTION
**Pull request by kcde**
``` javascript
// Check if human is playing against com
  useEffect(() => {
    if (opponent === 'COM' && !turn) comAi(turn, boardState, rules, handleClick, roundIsCompleted);
  }, [turn]);

  // play and allow next turn
  useEffect(() => {
    let isMounted = true;
    // check if all spots has some value but no winner, game result is a tie 
    if (isMounted && (winPattern.length < 1) && (boardState.every(spot => spot !== ''))) {
      setTies(prevState => prevState + 1);
      setFinalResult("It's a tie");
      resetGame();
    } else if (isMounted && boardState.some(spot => spot !== '')) play(); // next player's turn
    return () => {
      isMounted = false;
      clearTimeout(timeout);
    }
  }, [boardState]);

```

When Game.js mounts, following how the useEffects are called, the comAI plays and the "else if" statement in the second useEffect will evaluate to false which won't call the play() function that toggles the turn value. But comAi will cause boardState to update through the handleClick() function which will in effect cause a rerender of the component and since turn is still false, comAi will play again. By this second rerender, the "else if" statement in the second useEffect will evaluate to true which then calls the play() function.

so switching the order  of the useEffects like below  solved the issue

``` javascript
  // play and allow next turn
  useEffect(() => {
    let isMounted = true;
    // check if all spots has some value but no winner, game result is a tie 
    if (isMounted && (winPattern.length < 1) && (boardState.every(spot => spot !== ''))) {
      setTies(prevState => prevState + 1);
      setFinalResult("It's a tie");
      resetGame();
    } else if (isMounted && boardState.some(spot => spot !== '')) play(); // next player's turn
    return () => {
      isMounted = false;
      clearTimeout(timeout);
    }
  }, [boardState]);


// Check if human is playing against com
  useEffect(() => {
    if (opponent === 'COM' && !turn) comAi(turn, boardState, rules, handleClick, roundIsCompleted);
  }, [turn]);



```

### CODE REVIEW

**Comment by NARUDESIGNS** [comment]
I think your eslint or prettier code formatter is making some unnecessary changes which makes it hard to evaluate your fix.
Can you please revert changes, turn off your code formatter and push only changes that fixes the issue?

Thanks a bunch!

**Comment by kcde** [comment]
Alright

**Comment by NARUDESIGNS** [comment]
I just tried your changes on my local machine and it checks out properly so does your explanation on the rendering order. Thank you very much for your contribution. Please let me know when you've made changes to your PR and I can merge it as well as give you credits on my social media where I spoke about this issue. 
Thanks again!

**Comment by kcde** [comment]
Hi, 
Please confirm if the required changes have been effected

**Review by NARUDESIGNS — COMMENTED** [review]
Hi, I think your code formatter still made more changes.
You can use git show to see the changes before you push.
To resolve this simply 
* roll back all changes or better still create an entirely new PR in a new branch
* turn off eslint or prettier or whatever code formatter you use
* make changes (reorder the useEffect hook like you figured).
* use `git show` to review changes or if you use vscode, you can find this button in the image below and use it to view changes.
* Then push changes.

Screenshot 2022-04-21 at 12 53 59 AM

**Comment by kcde** [comment]
Okay, i was able to figure out the issue. vscode was removing whitespaces.
I did a git reset back to your last commit and made the change. I think It should only show my intended change now.
If this doesn't;t work I'll just make a new pr
thanks

File: src/pages/Game.js [hunk 1]
```diff
@@ -93,6 +88,13 @@ function Game({ opponent, marker, oppMarker, turn, setTurn, play }) {
     }
   }, [boardState]);
 
+    // Check if human is playing against com
+    useEffect(() => {
+      if (opponent === 'COM' && !turn) comAi(turn, boardState, rules, handleClick, roundIsCompleted);
+    }, [turn]);
+  
+    
+
```
Comment (NARUDESIGNS) [inline src/pages/Game.js]: Thank you very much @kcde looks great, just a little cleanup, please.
* Take out the 2 blank lines above this, the 3 blank lines are too much empty space in the code
* The second `useEffect()` hook (the one for comAi) is not properly indented. Just adjust it to be on the same line as the first `useEffect()` hook.

And that should be all. I appreciate your efforts and thanks again!

**Comment by kcde** [comment]
All fixed

**Comment by NARUDESIGNS** [comment]
> All fixed

Awesome!
Thank you so much!

**Pull request merged**